subsys/bluetooth/host/hci_core.c

```diff
@@ -2419,42 +2426,35 @@ static void hci_cmd_done(uint16_t opcode, uint8_t status, struct net_buf *evt_bu
 	}
 
 	/* Take the original command buffer reference. */
-	buf = atomic_ptr_clear((atomic_ptr_t *)&bt_dev.sent_cmd);
+	req = atomic_ptr_clear((atomic_ptr_t *)&bt_dev.sent_cmd);
 
-	if (!buf) {
+	if (!req) {
 		LOG_ERR("No command sent for cmd complete 0x%04x", opcode);
 		goto exit;
 	}
 
-	if (cmd(buf)->opcode != opcode) {
+	if (cmd(req)->opcode != opcode) {
 		LOG_ERR("OpCode 0x%04x completed instead of expected 0x%04x", opcode,
-			cmd(buf)->opcode);
-		buf = atomic_ptr_set((atomic_ptr_t *)&bt_dev.sent_cmd, buf);
-		__ASSERT_NO_MSG(!buf);
+			cmd(req)->opcode);
+		req = atomic_ptr_set((atomic_ptr_t *)&bt_dev.sent_cmd, req);
+		__ASSERT_NO_MSG(!req);
 		goto exit;
 	}
 
-	/* Response data is to be delivered in the original command
-	 * buffer.
-	 */
-	if (evt_buf != buf) {
-		net_buf_reset(buf);
-		bt_buf_set_type(buf, BT_BUF_EVT);
-		net_buf_reserve(buf, BT_BUF_RESERVE);
-		net_buf_add_mem(buf, evt_buf->data, evt_buf->len);
-	}
-
-	if (cmd(buf)->state && !status) {
-		struct bt_hci_cmd_state_set *update = cmd(buf)->state;
+	if (cmd(req)->state && !status) {
+		struct bt_hci_cmd_state_set *update = cmd(req)->state;
```

Comment:
Since this PR is doing some renamings, a few things pops out:

```c
/* Take the original command buffer reference. */
req = atomic_ptr_clear((atomic_ptr_t *)&bt_dev.sent_cmd);
```

So here we take a pointer to something called `sent_cmd` and store it in a variable called `req`, but `req` is not a `cmd`? Since we are doing `cmd(req)` then I assume `req` is something that wraps the command with other values. 

Should `sent_cmd` have been `sent_req`? 

Is the `rsp` buffer always a response (return parameters), or sometimes an event?

Reply:
Changed.  :)